RecoTracker/MkFitCore/src/MaterialEffects.h

```diff
@@ -19,10 +19,16 @@ namespace mkfit {
   public:
     MaterialEffects();
 
-    int getZbin(const float z) const { return (std::abs(z) * Config::nBinsZME) / (Config::rangeZME); }
-    int getRbin(const float r) const { return (r * Config::nBinsRME) / (Config::rangeRME); }
-    float getRlVal(const int zb, const int rb) const { return mRlgridME[zb][rb]; }
-    float getXiVal(const int zb, const int rb) const { return mXigridME[zb][rb]; }
+    int getZbin(const float z) const {
+      return std::isnan(z) ? -1 : (std::abs(z) * Config::nBinsZME) / (Config::rangeZME);
```

Comment:
Given the use of `-ffast-math`, I suppose this would be better with `edm::isFinite()`.